components/automate-deployment/pkg/target/local_target.go

```diff
@@ -787,20 +785,34 @@ func (t *LocalTarget) HabSupRestart(sortedServiceList []string) (bool, error) {
 	return true, errors.Wrap(err, "failed to restart habitat supervisor")
 }
 
-func (t *LocalTarget) waitForUnload(name string, timeout time.Duration) error {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
+func (t *LocalTarget) waitForUnload(ctx context.Context, name string) error {
+	errC := make(chan error)
 
-	for {
-		_, err := t.HabClient.ServiceInfo(ctx, name, "default")
-		if err == habapi.ErrServiceNotFound {
-			return nil
-		}
+	unloadService := func() {
+		for {
+			_, err := t.HabClient.ServiceInfo(ctx, name, "default")
+			if err == habapi.ErrServiceNotFound {
+				errC <- nil
+				break
+			}
 
-		if err != nil {
-			return err
+			if err != nil {
+				errC <- err
+				break
+			}
+
+			time.Sleep(500 * time.Millisecond)
 		}
-		time.Sleep(500 * time.Millisecond)
+	}
+
+	go unloadService()
```

Comment:
I'm not sure about refactoring this to use a goroutine. if the context timed out before, the "long running" ServiceInfo call would bail, so I think it was already correct.
With the goroutine version, I think we end up with cases where we can leak the goroutine. For example, if the unloadService function gets an error, and then the context times out and we return `ctx.Err()`, i think the goroutine will never be able to write to the err channel because there is no listener.